client/lib/analytics/ad-tracking/google-analytics-4.ts

```diff
@@ -0,0 +1,43 @@
+import { GaPurchase } from '../utils/jetpack-cart-to-ga-purchase';
+import { GaItem } from '../utils/jetpack-product-to-ga-item';
+import { TRACKING_IDS } from './constants';
+
+// Ensure setup has run.
+import './setup';
+
+function setup( params: Gtag.ConfigParams ) {
+	window.gtag( 'config', TRACKING_IDS.jetpackGoogleGA4Gtag, params );
+}
+
+function fireJetpackEcommercePurchase( purchase: GaPurchase ) {
+	window.gtag( 'event', 'purchase', {
+		send_to: TRACKING_IDS.jetpackGoogleGA4Gtag,
+		...purchase,
+	} );
+}
+
+function fireJetpackEcommerceAddToCart( item: GaItem ) {
+	window.gtag( 'event', 'add_to_cart', {
+		send_to: TRACKING_IDS.jetpackGoogleGA4Gtag,
+		value: item.price,
+		currency: 'USD',
+		items: [ item ],
+	} );
+}
+
+function firePageView( title: string, location: string, shouldSendToJetpack = false ) {
+	if ( shouldSendToJetpack ) {
+		window.gtag( 'event', 'page_view', {
+			send_to: TRACKING_IDS.jetpackGoogleGA4Gtag,
+			page_title: title,
+			page_location: location,
+		} );
+	}
+}
+
+export const GA4 = {
```

Comment:
That's a really good protip, thank you! Indeed, it's better to manage such things for the specific endpoints rather than to force a namespace approach everywhere.